src/test/java/org/jboss/aerogear/keycloak/metrics/PrometheusExporterTest.java

```diff
@@ -1,58 +1,226 @@
 import org.keycloak.events.Event;
 import org.keycloak.events.EventType;
+import org.keycloak.events.admin.AdminEvent;
 import org.keycloak.events.admin.OperationType;
+import org.keycloak.events.admin.ResourceType;
 
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
 
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.is;
+
+@SuppressWarnings("unchecked")
 public class PrometheusExporterTest {
-  private Event createEvent(EventType type) {
-    final Event event = new Event();
-    event.setType(type);
-    event.setRealmId("myrealm");
-    return event;
-  }
-
-  @Test
-  public void shouldRegisterAllKeycloakEvents() {
-    int userEvents = EventType.values().length;
-    int adminEvents = OperationType.values().length;
-
-    MatcherAssert.assertThat(
-      "All events registered",
-      userEvents + adminEvents == PrometheusExporter.counters.size());
-  }
-
-  @Test
-  public void shouldCorrectlyCountLogin() throws IOException {
-    final Event login = createEvent(EventType.LOGIN);
-    PrometheusExporter.instance().recordUserLogin(login);
-    assertLoggedInCount(1);
-    final Event logout = createEvent(EventType.LOGOUT);
-    PrometheusExporter.instance().recordUserLogout(logout);
-    assertLoggedInCount(0);
-  }
-
-  @Test
-  public void shouldCorrectlyCountImpersonate() throws IOException {
-    final Event imporsonate = createEvent(EventType.IMPERSONATE);
-    PrometheusExporter.instance().recordUserLogin(imporsonate);
-    assertLoggedInCount(1);
-    final Event logout = createEvent(EventType.LOGOUT);
-    PrometheusExporter.instance().recordUserLogout(logout);
-    assertLoggedInCount(0);
-  }
-
-  private void assertLoggedInCount(double number) throws IOException {
-    try (ByteArrayOutputStream stream = new ByteArrayOutputStream()) {
-      PrometheusExporter.instance().export(stream);
-      String result = new String(stream.toByteArray());
-      MatcherAssert.assertThat("Logout count",
-        result.contains("kc_logged_in_users{realm=\"myrealm\",} " + number));
-    }
-  }
-}
\ No newline at end of file
+
+    private static final String DEFAULT_REALM = "myrealm";
+
+    @Before
+    public void before() {
+        for (Counter counter : PrometheusExporter.instance().counters.values()) {
+            counter.clear();
+        }
+        PrometheusExporter.instance().totalLogins.clear();
+        PrometheusExporter.instance().totalFailedLoginAttempts.clear();
+        PrometheusExporter.instance().totalRegistrations.clear();
+    }
+
+    @Test
+    public void shouldRegisterCountersForAllKeycloakEvents() {
+        int userEvents = EventType.values().length;
+        int adminEvents = OperationType.values().length;
+
+        MatcherAssert.assertThat(
+            "All events registered",
+            userEvents + adminEvents - 3,                             // -3 comes from the events that
```

Comment:
This is sentence makes me feel uneasy :P
I think it would be better if we had the custom counters (namely logins, registrations and failures) in an array and we did `PrometheusExporter.instance().counters.size() + customCounters.size()`

Line 31-33 could also benefit from this:
```java
for (Counter counter in customCounters) {
    counter.clear();
}

Reply:
Plus, we could remove the comment

Reply:
@josemigallas I think that's possible. I've actually done it in https://github.com/aliok/keycloak-metrics-spi/compare/AEROGEAR-2096-mobile-metrics-tests...aliok:AEROGEAR-2096-mobile-metrics-tests-IMPROVEMENTS?expand=1 but didn't want to make this thing a bigger PR. I would say let's ignore this problem for now until I send the PR for that :)